### PR TITLE
Combat NPC recode

### PR DESCRIPTION
**Pull request by Rattlyy**
Recodes Combat NPCs to actually work. Tested with Sciath.

### CODE REVIEW

File: server/src/main/kotlin/net/horizonsend/ion/server/features/combatnpcs/CombatNPCs.kt [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import org.bukkit.event.player.PlayerQuitEvent
+import org.bukkit.inventory.EquipmentSlot
+import org.bukkit.inventory.ItemStack
+import org.bukkit.inventory.PlayerInventory
+import org.bukkit.inventory.meta.SkullMeta
+import java.util.*
+
+class CombatNPCs : Listener {
+
+	// CREDITS TO TRAINBOY
```
Comment (trainb0y) [inline server/src/main/kotlin/net/horizonsend/ion/server/features/combatnpcs/CombatNPCs.kt:32]: You don’t need to credit me like this lol

**Comment by Peter-Crawley** [comment]
Rebase required

File: server/src/main/kotlin/net/horizonsend/ion/server/features/combatnpcs/CombatNPCs.kt [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import org.bukkit.event.Listener
+import org.bukkit.event.entity.EntityDamageByEntityEvent
+import org.bukkit.event.entity.EntityDeathEvent
+import org.bukkit.event.player.PlayerJoinEvent
+import org.bukkit.event.player.PlayerQuitEvent
+import org.bukkit.inventory.EquipmentSlot
+import org.bukkit.inventory.ItemStack
+import org.bukkit.inventory.PlayerInventory
+import org.bukkit.inventory.meta.SkullMeta
+import java.util.*
```
Comment (Peter-Crawley) [inline server/src/main/kotlin/net/horizonsend/ion/server/features/combatnpcs/CombatNPCs.kt:28]: no wildcards

**Review by Peter-Crawley — CHANGES_REQUESTED** [review]

**Comment by Peter-Crawley** [comment]
merge conficts

**Pull request closed without merging**